--- branches/gcc-4_6-branch/gcc/ira-int.h	2012/09/21 10:08:35	191605
+++ branches/gcc-4_6-branch/gcc/ira-int.h	2012/09/21 10:09:04	191606
@@ -1123,8 +1123,13 @@
 ira_allocno_object_iter_cond (ira_allocno_object_iterator *i, ira_allocno_t a,
 			      ira_object_t *o)
 {
-  *o = ALLOCNO_OBJECT (a, i->n);
-  return i->n++ < ALLOCNO_NUM_OBJECTS (a);
+  int n = i->n++;
+  if (n < ALLOCNO_NUM_OBJECTS (a))
+    {
+      *o = ALLOCNO_OBJECT (a, n);
+      return true;
+    }
+  return false;
 }
 
 /* Loop over all objects associated with allocno A.  In each

