--- gdb-7.5.1_org/sim/common/gentmap.c	2012-05-19 18:46:14.000000000 +0200
+++ gdb-7.5.1/sim/common/gentmap.c	2021-08-14 14:56:42.000000000 +0200
@@ -2,6 +2,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 struct tdefs {
   char *symbol;
diff -NrU3 gdb-7.5.1_org/readline/rltty.c gdb-7.5.1/readline/rltty.c
--- gdb-7.5.1_org/readline/rltty.c	2011-05-12 01:38:39.000000000 +0200
+++ gdb-7.5.1/readline/rltty.c	2021-08-16 20:17:14.000000000 +0200
@@ -30,6 +30,7 @@
 #include <signal.h>
 #include <errno.h>
 #include <stdio.h>
+#include <sys/ioctl.h>
 
 #if defined (HAVE_UNISTD_H)
 #  include <unistd.h>
--- gdb-7.5.1_org/readline/terminal.c	2011-05-12 01:38:39.000000000 +0200
+++ gdb-7.5.1/readline/terminal.c	2021-08-16 21:01:55.000000000 +0200
@@ -47,6 +47,7 @@
 #endif
 
 #include <stdio.h>
+#include <sys/ioctl.h>
 
 /* System-specific feature definitions and include files. */
 #include "rldefs.h"
